webapp/views.py

```diff
@@ -1,139 +1,337 @@
+    if request.oauth.has_credentials():
+        return True
+    else: return False
+
+
+@decorators.oauth_enabled
+def addIfNewUser(request):
+    if request.oauth.has_credentials():
+       #print(request.oauth.credentials.id_token.items())
+       if User.objects.filter(username=request.oauth.credentials.id_token['email']).exists():
+           updateNameInfo=User.objects.get(username=request.oauth.credentials.id_token['email'])
+           #print('user exists') 
+		   ## Update the users name ( in the case that the user was created through group/role management pages)
+           updateNameInfo.set_first_name=request.oauth.credentials.id_token['given_name']
+           updateNameInfo.set_last_name=request.oauth.credentials.id_token['family_name']
+           updateNameInfo.save()         
+       else:
+	   #Create a new user from the logged in google info
+         newUser=User.objects.create_user(request.oauth.credentials.id_token['email'],request.oauth.credentials.id_token['email'],'tempPass')
+         newUser.first_name=request.oauth.credentials.id_token['given_name']
+         newUser.last_name=request.oauth.credentials.id_token['family_name']
+         newUser.save()
+       user = authenticate(username=request.oauth.credentials.id_token['email'], password='tempPass')
+       if user is not None:
+           #print('logging in user')
+           login(request, user)
+
+def logout_view(request):
+   logout(request)
+   #return HttpResponseRedirect('login')
+   return render(
+            request,
+            'webapp/logout.html',
+            {
+                'title':'Syndication Tool',
+                'year':datetime.now().year,
+                'isLoggedOut':True
+            }
+        )
+
+@login_required(login_url='/eventsyndication/login')
+#@permission_required('webapp.CanLogin', login_url='/eventsyndication/logout')
+def home(request,error=''):
+    if request.user.has_perm('webapp.CanLogin'):
+        """Renders the home page."""
+        assert isinstance(request, HttpRequest)
+        return render(
+            request,
+            'webapp/index.html',
+            {
+                'title':'Syndication Tool',
+                'year':datetime.now().year,
+                'errorMessage':error
+            }
+        )
+    else:
+        return render(
+            request,
+            'webapp/logout.html',
+            {
+                'title':'Syndication Tool',
+                'year':datetime.now().year,
+                'errorMessage':"You don't have permission to be logged in",
+                'isLoggedOut':True
+            }
+        )
+
+
+
+#@permission_required('webapp.CreatePage_View', login_url='/eventsyndication/')
 def createEvent(request):
+  if request.user.has_perm('webapp.CreatePage_View'):  
     """Renders the createEvent page."""
     assert isinstance(request, HttpRequest)
     form = AddEventForm()
+    #print (request.user.has_perm('webapp.CreatePage_Action'))

```

Comment:
can we clean up some of these comments?  If we have to leave any in, I'd like there to be another comment explaining why, because it's not immediately clear to me (and probably won't be to whoever we might have to hand the project off to)